### PR TITLE
mrc-3328 Keep user's language preference outside project state

### PR DESCRIPTION
**Pull request by LekanAnni**
## Description

This removes language state from getting stored in the backend and local storage. Add new key to local storage and monitor language alongside language state. The instruction on the ticket is a lot miss leading, I have been trying to solve this differently, trying to remove language from the state completely. 

We still need language to be in the state for reusability across the app, This PR keeps track of user language outside of projects, with additional key in local storage. 

## Type of version change
_Delete as appropriate_

Major - Minor - Patches - None

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/localStorageManager.ts [hunk 1]
```diff
@@ -64,6 +63,16 @@ declare const currentUser: string;
 
 export class LocalStorageManager {
 
+    saveLanguageState = (lang: Language) => {
+        localStorage.setItem("language", lang);
+    }
+
+    getLanguageState = (): Language => {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/localStorageManager.ts:70]:
```suggestion
    getLanguage = (): Language => {
```

File: src/app/static/src/app/localStorageManager.ts [hunk 2]
```diff
@@ -64,6 +63,16 @@ declare const currentUser: string;
 
 export class LocalStorageManager {
 
+    saveLanguageState = (lang: Language) => {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/localStorageManager.ts:66]:
```suggestion
    saveLanguage = (lang: Language) => {
```

File: src/app/static/src/app/localStorageManager.ts [hunk 3]
```diff
@@ -83,7 +92,12 @@ export class LocalStorageManager {
         const appStateKey = getAppStateKey(dataExplorationMode);
         const item = window.localStorage.getItem(appStateKey);
         if (item) {
-            return JSON.parse(item) as Partial<RootState>;
+            const partialRootState = JSON.parse(item) as Partial<RootState>;
+            /**
+             * For smooth compatibility, code uses local storage persisted language state,
+             * and default to English if needed.
+             */
+            return {...partialRootState, ...{language: this.getLanguageState()}}
```
Comment (EmmaLRussell) [inline src/app/static/src/app/localStorageManager.ts:100]:
```suggestion
            return {...partialRootState, language: this.getLanguageState()}
```

**Review by EmmaLRussell — COMMENTED** [review]
Looks great. Correctly reloads a project with the user's new language preference if it has changed since the project was saved. I think leaving the state value in place but loading it from local storage is a good call for minimising the code changes, 

One thing it doesn't do, which was suggested in the ticket, was to save the user's language preference from the Forgot Password page. But I think that;s not a high priority.

**Review by M-Kusumgar — CHANGES_REQUESTED** [review]
Code looks good, little bit confused about functionality. Now the language is just updated in local storage so the app doesn't update when you select a new language from the dropdown? Think we do need to retain that behaviour because currently if i select "fr" from the dropdown, nothing actually happens until i refresh the page

**Review by M-Kusumgar — APPROVED** [review]

File: src/app/static/src/app/localStorageManager.ts (hunk 1 above)
Comment (LekanAnni) [reply src/app/static/src/app/localStorageManager.ts]: Done.

File: news.md.save [hunk 4]
```diff
@@ -0,0 +1,1026 @@
+# hint 2.39.0
```
[the remark this answers is not in the document]
Comment (LekanAnni) [reply news.md.save]: Done

File: src/app/static/src/app/localStorageManager.ts (hunk 2 above)
Comment (LekanAnni) [reply src/app/static/src/app/localStorageManager.ts]: Done

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**